WordPress/src/main/java/org/wordpress/android/ui/blaze/ui/blazeoverlay/BlazeOverlayFragment.kt

```diff
@@ -0,0 +1,282 @@
+                val postModel by viewModel.promoteUiState.observeAsState(BlazeUiState.PromoteScreen.Site)
+                LocaleAwareComposable(
+                    locale = LocaleManager.languageLocale(userLanguage),
+                    onLocaleChange = viewModel::setAppLanguage
+                ) {
+                    viewModel.trackOverlayDisplayed()
+                    BlazeOverlayContent(postModel)
+                }
+            }
+        }
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    @Composable
+    fun BlazeOverlayContent(
+        postModelState: BlazeUiState.PromoteScreen,
+        isDarkTheme: Boolean = false
+    ) {
+        val post = when (postModelState) {
+            is BlazeUiState.PromoteScreen.PromotePost -> postModelState.postUIModel
+            else -> null
+        }
+        Column(
+            horizontalAlignment = Alignment.CenterHorizontally,
+            modifier = Modifier.padding(
+                top = Margin.ExtraLarge.value,
+                start = Margin.ExtraLarge.value,
+                end = Margin.ExtraLarge.value
+            )
+        ) {
+            Image(
+                painterResource(id = R.drawable.ic_blaze_promotional_image),
+                contentDescription = stringResource(id = R.string.blaze_activity_title),
+                modifier = Modifier.size(100.dp)
+            )
+            Text(
+                stringResource(id = R.string.blaze_promotional_text),
+                fontSize = FontSize.DoubleExtraLarge.value,
+                fontWeight = FontWeight.Bold,
+                modifier = Modifier.padding(top = Margin.ExtraExtraMediumLarge.value),
+                textAlign = TextAlign.Center
+            )
+            Subtitles(
+                listOf(
+                    R.string.blaze_promotional_subtitle_1,
+                    R.string.blaze_promotional_subtitle_2,
+                    R.string.blaze_promotional_subtitle_3
+                ),
+                modifier = Modifier.padding(start = 20.dp, end = 20.dp)
+            )
+            post?.let {
+                PostThumbnailView(
+                    it, modifier = Modifier
+                        .fillMaxWidth()
+                        .wrapContentHeight()
+                )
+            }
+
+            ImageButton(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(48.dp)
+                    .padding(start = 20.dp, end = 20.dp)
+                    .background(
+                        Color.Black,
+                        shape = RoundedCornerShape(15.dp)
+                    ),
+                buttonText = UiString.UiStringRes(R.string.blaze_post_promotional_button),
+                drawableRight = Drawable(R.drawable.ic_promote_with_blaze),
+                onClick = { viewModel.onPromoteWithBlazeClicked() }
+            )
+        }
+    }
+
+    @Preview
+    @Composable
+    fun PreviewThumbNailView() {
```

Comment:
np: Since `thumbnail` is one word the `n` doesn't need capitalisation.
Also preview composables can alway be `private` 👍

Reply:
Yup, that Makes sense. Done in 772aeeedf058c865faf78389e898675777b1cb38